apps/rewards/src/invite/rewardSender.service.ts

```diff
@@ -0,0 +1,203 @@
+import { KomenciLoggerService } from '@app/komenci-logger'
+import { ContractKit } from '@celo/contractkit'
+import { Inject, Injectable } from '@nestjs/common'
+import { Cron, CronExpression } from '@nestjs/schedule'
+import BigNumber from 'bignumber.js'
+import { EntityManager, In, Raw, Repository } from 'typeorm'
+import { appConfig, AppConfig } from '../config/app.config'
+import { RelayerProxyService } from '../relayer/relayer_proxy.service'
+import { InviteReward, RewardStatus } from './inviteReward.entity'
+import { InviteRewardRepository } from './inviteReward.repository'
+
+const WEI_PER_UNIT = 1000000000000000000
+const MAX_INVITES_PER_QUERY = 100
+const REWARD_VALUE_CACHE_DURATION = 1 * 60 * 1000 // One minute
+
+interface WatchedInvite {
+  inviteId: string
+  txHash: string
+  sentAt: number
+  markAsDeadLetterIfFailed: boolean
```

Comment:
It might be more flexible to change this to an attempt count and then dead-letter after a configurable number of attempts